service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/BlobCopier.java

```diff
@@ -0,0 +1,109 @@
+package bio.terra.workspace.service.resource.controlled.cloud.azure;
+
+import static java.util.stream.Collectors.groupingBy;
+
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.resource.controlled.cloud.azure.storage.ControlledAzureStorageResource;
+import bio.terra.workspace.service.resource.controlled.cloud.azure.storageContainer.ControlledAzureStorageContainerResource;
+import com.azure.core.util.polling.PollResponse;
+import com.azure.core.util.polling.SyncPoller;
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.models.BlobCopyInfo;
+import com.azure.storage.blob.models.BlobItem;
+import java.time.Duration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BlobCopier {
+  private final Logger logger = LoggerFactory.getLogger(BlobCopier.class);
+  private final AzureStorageAccessService storageAccessService;
+  private final AuthenticatedUserRequest userRequest;
+
+  private static final Duration MAX_POLL_TIMEOUT = Duration.ofMinutes(10);
```

Comment:
We can tweak this as needed, 10 minutes seems generous but I'm happy to bump it up / down if folks have thoughts.

Reply:
Do I understand you are allowing only 10 minutes for copying all of the blobs? (Seems short) Or is it copying a single blob? (Not sure)

On GCP when we copy across regions, the copy times could be very long. And a local copy was not an actual copy - just some GCS backend operation. So that was very fast. Do you know what this copy is doing under the covers?

Reply:
feels like a reasonable place to start to me